### PR TITLE
fix: parse the process name to get the process type instead of using port count

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/7393

Get the process type by parsing the name instead of using port count.

### CODE REVIEW

**Comment by ChanYiLin** [comment]
E2E testing (-m coretest): 
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/6372/console

**Comment by ChanYiLin** [comment]
E2E test **PASSED**

File: pkg/instance/instance.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 
 	"github.com/longhorn/longhorn-instance-manager/pkg/client"
 	"github.com/longhorn/longhorn-instance-manager/pkg/meta"
+	improc "github.com/longhorn/longhorn-instance-manager/pkg/process"
```
Comment (innobead) [inline pkg/instance/instance.go:21]: like the convention for the alias for our own lib. `**im**proc`

File: pkg/process/process.go [hunk 2]
```diff
@@ -186,3 +187,11 @@ func (p *Process) IsStopped() bool {
 	defer p.lock.RUnlock()
 	return p.State == StateStopped || p.State == StateError
 }
+
+func IsEngineProcess(name string) bool {
+	// engine process name example: pvc-5a8ee916-5989-46c6-bafc-ddbf7c802499-e-0
+	nameSlices := strings.Split(name, "-")
+	processType := nameSlices[len(nameSlices)-2]
+
+	return processType == "e"
```
Comment (innobead) [inline pkg/process/process.go:196]: Use regex instead. Also, probably can add to common lib if appropriate, as it is used in your another longhorn manager PR. 

Suggest adding a doc comment for this function to better explain what the arg means.

**Review by innobead — DISMISSED** [review]
In genreal, LGTM. Just a few feedback.

File: pkg/process/process.go (hunk 2 above)
Comment (ChanYiLin) [reply pkg/process/process.go:196]: PR: https://github.com/longhorn/go-common-libs/pull/25

**Comment by innobead** [comment]
Please fix the conflict and then we can move forward.

**Comment by ChanYiLin** [comment]
Hi @innobead 
Fixed the conflicts, thanks

File: pkg/instance/instance.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ import (
 	spdkapi "github.com/longhorn/longhorn-spdk-engine/pkg/api"
 	spdkclient "github.com/longhorn/longhorn-spdk-engine/pkg/client"
 
+	lhutils "github.com/longhorn/go-common-libs/utils"
```
Comment (derekbit) [inline pkg/instance/instance.go:19]: newline for separation

```suggestion
	lhutils "github.com/longhorn/go-common-libs/utils"
	
```

File: pkg/process/process_manager.go [hunk 4]
```diff
@@ -155,13 +155,13 @@ func (pm *Manager) getProcessesToUpdateConditions(volumeMountPointMap map[string
 
 	for _, p := range pm.processes {
 		p.lock.Lock()
-		if isEngineProcess(p) && p.State == StateRunning {
+		if lhUtils.IsEngineProcess(p.Name) && p.State == StateRunning {
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply pkg/process/process_manager.go:158]: yeah, thanks for the finding
Do you have any good regex pattern for this case?

File: pkg/instance/instance.go (hunk 3 above)
Comment (ChanYiLin) [reply pkg/instance/instance.go]: Fixed, thanks

File: pkg/process/process_manager.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply pkg/process/process_manager.go:158]: This could work: ``.+?-e-[^-]*\d$``

[the remark this answers is not in the document]
Comment (derekbit) [reply pkg/process/process_manager.go:158]: LGTM!

File: pkg/instance/instance.go [hunk 5]
```diff
@@ -21,6 +21,8 @@ import (
 	"github.com/longhorn/longhorn-instance-manager/pkg/types"
 
 	rpc "github.com/longhorn/longhorn-instance-manager/pkg/imrpc"
+
+	lhutils "github.com/longhorn/go-common-libs/utils"
```
Comment (derekbit) [inline pkg/instance/instance.go:25]:
```
	"context"
	"fmt"
	"io"
	"time"

	"github.com/pkg/errors"
	"github.com/sirupsen/logrus"
	"golang.org/x/sync/errgroup"
	"grpccodes "google.golang.org/grpc/codes"
	"grpcstatus "google.golang.org/grpc/status"
	"google.golang.org/protobuf/types/known/emptypb"

	lhutils "github.com/longhorn/go-common-libs/utils"
	spdkapi "github.com/longhorn/longhorn-spdk-engine/pkg/api"
	spdkclient "github.com/longhorn/longhorn-spdk-engine/pkg/client"

	"github.com/longhorn/longhorn-instance-manager/pkg/client"
	"github.com/longhorn/longhorn-instance-manager/pkg/meta"
	"github.com/longhorn/longhorn-instance-manager/pkg/types"

	rpc "github.com/longhorn/longhorn-instance-manager/pkg/imrpc"
```

Comment (ChanYiLin) [reply pkg/instance/instance.go]: Fixed.

**Comment by ChanYiLin** [comment]
HI @innobead @derekbit 
All fixed. Thanks

**Review by derekbit — APPROVED** [review]
LGTM

**Review by derekbit — APPROVED** [review]

**Pull request merged**